apps/opencs/model/prefs/boolsetting.cpp

```diff
@@ -22,6 +22,8 @@ CSMPrefs::BoolSetting& CSMPrefs::BoolSetting::setTooltip (const std::string& too
 
 std::pair<QWidget *, QWidget *> CSMPrefs::BoolSetting::makeWidgets (QWidget *parent)
 {
+    if (mWidget != nullptr)
+        delete mWidget;
```

Comment:
Does makeWidgets run more than once?

Reply:
As I pointed in the first message, I'm stiil a newbie here. In my personal projects I'm very strict against memory allocations. Maybe in future there will be tons of makeWidgets calls. :smile: If there is a possibility of a memory leak, then it can happen sometimes.

Reply:
If you run makeWidgets, then what are you planning to do with the old ones? Do we need to disconnect the widget first from Qt? Would that too be a memory leak to leave a connection to a deleted pointer?

Reply:
FWIW, it's safe to call `delete` with a nullptr.
```
if(x != nullptr)
    delete x;
x = nullptr;
```
is a redundant null check.

Reply:
AFAIK this is wrong you never delete any QWidget, they are deleted by their parent.

Reply:
This change is indeed wrong. The mWidget variable does not own the Widget. It is managed outside of this class. Please revert.